### PR TITLE
feat(cli): add client side service proxy generation for `lb4 openapi` command

### PR DESCRIPTION
**Pull request by raymondfeng**
This PR extends `lb4 openapi` with client-side service proxy code generation. The primary use case is to generate strongly-typed service proxies for REST apis that have Swagger 2.0 or OpenAPI 3.0 spec.

Usage:

`lb4 openapi --client`

It will prompt the user to give a datasource name and generate:

- models (models and types)
- services (for strongly-typed service providers per each tag)
- datasources (for openapi connector, can be supplied by `--datasource`)

The `--client` can be used with `--server`. By default, `client` is `false` and `server` is true. 

See docs at https://github.com/strongloop/loopback-next/blob/cli-openapi-client/docs/site/OpenAPI-generator.md

Depends on https://github.com/strongloop/loopback-connector-openapi/pull/10

## Checklist

👉 [Read and sign the CLA (Contributor License Agreement)](https://cla.strongloop.com/agreements/strongloop/loopback-next) 👈

- [x] `npm test` passes on your machine
- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [x] API Documentation in code was updated
- [x] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [x] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `examples/*` were updated

👉 [Check out how to submit a PR](https://loopback.io/doc/en/lb4/submitting_a_pr.html) 👈

### CODE REVIEW

File: packages/cli/snapshots/integration/generators/openapi-petstore.integration.snapshots.js [hunk 1]
```diff
@@ -7,42 +7,52 @@
...
-
-Sed tempus felis lobortis leo pulvinar rutrum. Nam mattis velit nisl, eu condimentum ligula luctus nec. Phasellus semper velit eget aliquet faucibus. In a mattis elit. Phasellus vel urna viverra, condimentum lorem id, rhoncus nibh. Ut pellentesque posuere elementum. Sed a varius odio. Morbi rhoncus ligula libero, vel eleifend nunc tristique vitae. Fusce et sem dui. Aenean nec scelerisque tortor. Fusce malesuada accumsan magna vel tempus. Quisque mollis felis eu dolor tristique, sit amet auctor felis gravida. Sed libero lorem, molestie sed nisl in, accumsan tempor nisi. Fusce sollicitudin massa ut lacinia mattis. Sed vel eleifend lorem. Pellentesque vitae felis pretium, pulvinar elit eu, euismod sapien.
-
-   * 
-
+Nam sed condimentum est. Maecenas tempor sagittis sapien, nec rhoncus sem
```
Comment (emonddr) [inline packages/cli/snapshots/integration/generators/openapi-petstore.integration.snapshots.js:32]: why do we have a big latin paragraph?

File: packages/cli/snapshots/integration/generators/openapi-uspto-with-anonymous.integration.snapshots.js [hunk 2]
```diff
@@ -131,7 +164,7 @@ export type PerformSearchRequestBodyWithRelations = PerformSearchRequestBody & P
 `;
 
 
-exports[`openapi-generator specific files generates all the proper files 4`] = `
+exports[`openapi-generator uspto with anonymous generates all the proper files 6`] = `
```
Comment (emonddr) [inline packages/cli/snapshots/integration/generators/openapi-uspto-with-anonymous.integration.snapshots.js:167]: `uspto` ?

File: docs/site/OpenAPI-generator.md [hunk 3]
```diff
@@ -370,6 +392,146 @@ export class AccountController {
...
+   * application to be shut down gracefully.
+   */
+  stop(): ValueOrPromise<void> {
+    return super.disconnect();
+  }
+}
+```
+
+For each tag, a service interface and proxy are generated to provide
+strongly-typed access to the endpoint.
```
Comment (jannyHou) [inline docs/site/OpenAPI-generator.md:456]: nitpick IIUC a tag contains multiple endpoints
```suggestion
strongly-typed access to the endpoints.
```

File: packages/cli/snapshots/integration/generators/openapi-petstore.integration.snapshots.js (hunk 1 above)
Comment (raymondfeng) [reply packages/cli/snapshots/integration/generators/openapi-petstore.integration.snapshots.js:32]: That's from one of the openapi specs we use for testing.

File: packages/cli/snapshots/integration/generators/openapi-uspto-with-anonymous.integration.snapshots.js (hunk 2 above)
Comment (raymondfeng) [reply packages/cli/snapshots/integration/generators/openapi-uspto-with-anonymous.integration.snapshots.js:167]: US Patent Office

**Pull request merged**